### PR TITLE
Update HTTP docs for Godot 4

### PR DESCRIPTION
**Pull request by mhilbrunner**
Supersedes https://github.com/godotengine/godot-docs/pull/7049, https://github.com/godotengine/godot-docs/pull/7083

- Godot 4 API changes (signal connections, request function has different params)
  - Prefer to use non-string based connections
- Changed the URL used for the example, using the public Github API instead of `http://www.mocky.io/`
- Added intro text with a general overview and relevant links
- Simplified the example by removing the button & corresponding UI/editor setup and linking
- Reordered content

### CODE REVIEW

File: tutorials/networking/http_request_class.rst [hunk 1]
```diff
@@ -1,70 +1,91 @@
...
-Create a new empty scene, add a CanvasLayer as the root node and add a script to it. Then add two child nodes to it: a Button and an HTTPRequest node. You will need to connect the following signals to the CanvasLayer script:
+Preparing the scene
+-------------------
 
-- Button.pressed: When the button is pressed, we will start the request.
-- HTTPRequest.request_completed: When the request is completed, we will get the requested data as an argument.
+Create a new empty scene, add a root :ref:`Node <class_Node>` and add a script to it.
+Then add a :ref:`HTTPRequest <class_HTTPRequest>` node as a child.
 
 .. image:: img/rest_api_scene.png
```
Comment (Piralein) [inline tutorials/networking/http_request_class.rst:54]: The image still shows the CanvasLayer/Button setup. It should be updated to match the new description.

Comment (mhilbrunner) [reply tutorials/networking/http_request_class.rst:54]: Damn, you're a life saver - quite right, not sure how I missed this one in my local build 😀

File: tutorials/networking/http_request_class.rst [hunk 2]
```diff
@@ -1,70 +1,91 @@
...
+`HTTP requests <https://developer.mozilla.org/en-US/docs/Web/HTTP>`_ are useful
+to communicate with web servers and other non-Godot programs.
+
+Compared to Godot's other networking features (like
+:ref:`High-level multiplayer <doc_high_level_multiplayer>`),
+HTTP requests have more overhead and take more time to get going,
+so they aren't suited for real-time communication, and aren't great to send
+lots of small updates as is common for multiplayer gameplay.
+
+HTTP however offers interoperability with external
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:18]:
```suggestion
HTTP, however, offers interoperability with external
```

File: tutorials/networking/http_request_class.rst [hunk 3]
```diff
@@ -1,70 +1,91 @@
...
+HTTP requests in Godot
+----------------------
+
 The :ref:`HTTPRequest <class_HTTPRequest>` node is the easiest way to make HTTP requests in Godot.
-It is backed by the more low-level :ref:`HTTPClient <class_HTTPClient>`, for which a tutorial is available :ref:`here <doc_http_client_class>`.
+It is backed by the more low-level :ref:`HTTPClient <class_HTTPClient>`,
+for which a tutorial is available :ref:`here <doc_http_client_class>`.
 
-For the sake of this example, we will create a simple UI with a button, that when pressed will start the HTTP request to the specified URL.
+For this example, we will make a HTTP request to GitHub to retrieve the name
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:38]:
```suggestion
For this example, we will make an HTTP request to GitHub to retrieve the name
```

File: tutorials/networking/http_request_class.rst [hunk 4]
```diff
@@ -1,70 +1,91 @@
...
+of the latest Godot release.
 
 .. warning::
 
-    When exporting to Android, make sure to enable the ``INTERNET``
+    When exporting to **Android**, make sure to enable the **Internet**
     permission in the Android export preset before exporting the project or
     using one-click deploy. Otherwise, network communication of any kind will be
     blocked by Android.
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:46]:
```suggestion
    blocked by the Android OS.
```
Reads a bit better and is probably more specific.

File: tutorials/networking/http_request_class.rst [hunk 5]
```diff
@@ -1,70 +1,91 @@
...
----------------
-
-Create a new empty scene, add a CanvasLayer as the root node and add a script to it. Then add two child nodes to it: a Button and an HTTPRequest node. You will need to connect the following signals to the CanvasLayer script:
+Preparing the scene
+-------------------
 
-- Button.pressed: When the button is pressed, we will start the request.
-- HTTPRequest.request_completed: When the request is completed, we will get the requested data as an argument.
+Create a new empty scene, add a root :ref:`Node <class_Node>` and add a script to it.
+Then add a :ref:`HTTPRequest <class_HTTPRequest>` node as a child.
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:52]:
```suggestion
Then add an :ref:`HTTPRequest <class_HTTPRequest>` node as a child.
```

File: tutorials/networking/http_request_class.rst [hunk 6]
```diff
@@ -1,70 +1,91 @@
...
 
 .. image:: img/rest_api_scene.png
 
-Scripting
----------
+Scripting the request
+---------------------
 
-Below is all the code we need to make it work. The URL points to an online API mocker; it returns a predefined JSON string, which we will then parse to get access to the data.
+When the project is started (so in ``_ready()``), we're going to send a HTTP request
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:59]:
```suggestion
When the project is started (so in ``_ready()``), we're going to send an HTTP request
```

File: tutorials/networking/http_request_class.rst [hunk 7]
```diff
@@ -74,58 +95,60 @@ Below is all the code we need to make it work. The URL points to an online API m
             }
         }
 
-With this, you should see ``(hello:world)`` printed on the console; ``hello`` being a key, ``world`` being a value, and both of them are of type :ref:`String <class_string>`.
-
+Save the script and the scene and run the project.
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:98]:
```suggestion
Save the script and the scene, and run the project.
```

File: tutorials/networking/http_request_class.rst [hunk 8]
```diff
@@ -74,58 +95,60 @@ Below is all the code we need to make it work. The URL points to an online API m
...
+The name of the most recent Godot release on Github should be printed to the console.
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:99]:
```suggestion
The name of the most recent Godot release on Github should be printed to the output log.
```

File: tutorials/networking/http_request_class.rst [hunk 9]
```diff
@@ -74,58 +95,60 @@ Below is all the code we need to make it work. The URL points to an online API m
...
-
-Sending data to server
-----------------------
+Setting custom HTTP headers
+---------------------------
 
-Until now, we have limited ourselves to requesting data from a server. But what if you need to send data to the server? Here is a common way of doing it:
+Of course, you can also set custom HTTP headers. These are given as a string array,
+with each string containing a header in the format ``"header: value"``.
+For example, to set a custom user agent (the HTTP ``user-agent`` header) you could use the following:
```
Comment (YuriSizov) [inline tutorials/networking/http_request_class.rst:136]: I think by convention headers are supposed to be capitalized, and you capitalized `Content-Type` above. So for consistency I'd do it here (and the example below) as well. Can also do it to the format, e.g. `Header: value`.

**Review by YuriSizov — APPROVED** [review]
Some nits, great update overall :)

File: tutorials/networking/http_request_class.rst (hunk 9 above)
Comment (mhilbrunner) [reply tutorials/networking/http_request_class.rst:136]: Ah, this was just how it was before my changes.
While the RFC spec defines them as case insensitive, I agree :) Its common convention, and we indeed use that elsewhere.

**Comment by mhilbrunner** [comment]
Pushed fixes for the feedback, and updated the image. :) Thanks everyone!

**Pull request merged**